### PR TITLE
Match some 16F740 funcs

### PR DESCRIPTION
**Pull request by ThreeMustyFears**

### CODE REVIEW

**Comment by ThreeMustyFears** [comment]
I know that the style is to put the variables using globals at the top but for btl_state_update_change_partner I had to put gPlayerData below a couple others as it fixed the ordering for loading partnerActor and playerActor.

**Comment by BowserSlug** [comment]
🚀🚀🚀🚀 This PR matches 4 functions (+0.14%) on `us`.

✅ This PR fixes 30 warnings!

File: src/16F740.c [hunk 1]
```diff
@@ -224,7 +228,205 @@ void btl_state_update_1C(void) {
...
+                script = start_script(partner->onTurnChanceScriptSource, 0xA, 0);
+                partner->onTurnChangeScript = script;
+                partner->onTurnChangeID = script->id;
+                script->owner1.enemyID = 0x100;
+                gBattleState2 = 1;
+            }
+
+            break;
+        case 1:
+            if (does_script_exist(partner->onTurnChangeID) == FALSE) {
```
Comment (nanaian) [inline src/16F740.c:263]:
```suggestion
            if (!does_script_exist(partner->onTurnChangeID)) {
```

File: src/16F740.c [hunk 2]
```diff
@@ -224,7 +228,205 @@ void btl_state_update_1C(void) {
...
+            }
+
+            battleStatus->unk_8C = 0;
+            if (battleStatus->outtaSightActive == FALSE) {
+                gBattleState2 = 5;
+            } else {
+                if (battleStatus->outtaSightActive > 0) {
+                    D_8029F254 = 1;
+                }
+                battleStatus->battlePhase = 0xC;
```
Comment (nanaian) [inline src/16F740.c:253]: I believe there's an enum for this

File: src/16F740.c [hunk 3]
```diff
@@ -224,7 +228,205 @@ void btl_state_update_1C(void) {
...
+
+    switch (gBattleState2) {
+        case 0:
+            player->flags &= ~(ACTOR_FLAG_8000000 | ACTOR_FLAG_4000000);
+            if (partner != NULL) {
+                partner->flags &= ~(ACTOR_FLAG_8000000 | ACTOR_FLAG_4000000);
+            }
+
+            battleStatus->unk_8C = 0;
+            if (battleStatus->outtaSightActive == FALSE) {
```
Comment (nanaian) [inline src/16F740.c:247]: I think `outtaSightActive` is the number of turns
```suggestion
            if (!battleStatus->outtaSightActive == 0) {
```

File: src/16F740.c [hunk 4]
```diff
@@ -224,7 +228,205 @@ void btl_state_update_1C(void) {
...
+            }
+            gBattleState2 = 0xB;
+            gBattleStatus.flags2 &= ~BS_FLAGS2_2;
+            gBattleStatus.flags2 &= ~BS_FLAGS2_4;
+            gBattleStatus.flags2 &= ~BS_FLAGS2_8;
+            gBattleStatus.flags2 &= ~BS_FLAGS2_10;
+
+            gBattleStatus.flags1 &= ~BS_FLAGS1_8;
+            if (player->koStatus == 0xD) {
+                dispatch_event_player(0x34);
```
Comment (nanaian) [inline src/16F740.c:284]: Use enum

File: src/16F740.c [hunk 5]
```diff
@@ -224,7 +228,205 @@ void btl_state_update_1C(void) {
...
+            player->staticStatus = 0;
+            player->stoneStatus = 0;
+            player->transStatus = 0;
+            player->koStatus = 0;
+            player->koDuration = 0;
+            player->ptrDefuffIcon->ptrPropertyList[15] = 0;
+
+            if (partner != NULL) {
+                if (partner->koStatus == 0xD) {
+                    dispatch_event_partner(0x34);
```
Comment (nanaian) [inline src/16F740.c:297]: Use enum

File: src/16F740.c [hunk 6]
```diff
@@ -224,7 +228,205 @@ void btl_state_update_1C(void) {
...
+                partner->staticStatus = 0;
+                partner->stoneStatus = 0;
+                partner->transStatus = 0;
+                partner->koStatus = 0;
+                partner->koDuration = 0;
+                partner->ptrDefuffIcon->ptrPropertyList[15] = 0;
+            }
+            break;
+        case 10:
+            if (player->onHitScript == NULL || does_script_exist(player->onHitID) == FALSE) {
```
Comment (nanaian) [inline src/16F740.c:310]:
```suggestion
            if (player->onHitScript == NULL || !does_script_exist(player->onHitID)) {
```

File: src/16F740.c [hunk 7]
```diff
@@ -224,7 +228,205 @@ void btl_state_update_1C(void) {
...
+                player->onHitScript = NULL;
+                if (partner != NULL) {
+                    if (partner->onHitScript == NULL || does_script_exist(partner->onHitID) == FALSE) {
```
Comment (nanaian) [inline src/16F740.c:313]:
```suggestion
                    if (partner->onHitScript == NULL || !does_script_exist(partner->onHitID)) {
```

File: src/16F740.c [hunk 8]
```diff
@@ -267,7 +469,81 @@ INCLUDE_ASM(s32, "16F740", btl_state_update_run_away);
...
+                    player->takeTurnID = script->id;
+                    script->owner1.enemyID = 0;
+                    gBattleState2 = 2;
+                } else {
+                    gBattleState2 = 0xA;
+                }
+            }
+            break;
+        case 2:
+            if (does_script_exist(player->takeTurnID) == FALSE && battleStatus->unk_8C == 0) {
```
Comment (nanaian) [inline src/16F740.c:533]:
```suggestion
            if (!does_script_exist(player->takeTurnID) && battleStatus->unk_8C == 0) {
```

File: src/16F740.c [hunk 9]
```diff
@@ -267,7 +469,81 @@ INCLUDE_ASM(s32, "16F740", btl_state_update_run_away);
...
+            if (partner != NULL) {
+                script = start_script(partner->takeTurnScriptSource, 0xA, 0);
+                partner->takeTurnScript = script;
+                partner->takeTurnID = script->id;
+                script->owner1.enemyID = 0x100;
+            }
+            gBattleState2 = 1;
+            break;
+        case 1:
+            if (does_script_exist(player->takeTurnID) == FALSE && battleStatus->unk_8C == 0) {
```
Comment (nanaian) [inline src/16F740.c:519]:
```suggestion
            if (!does_script_exist(player->takeTurnID) && battleStatus->unk_8C == 0) {
```

File: src/16F740.c [hunk 10]
```diff
@@ -282,7 +558,132 @@ ApiStatus DisablePartnerBlur(Evt* script, s32 isInitialCall) {
...
+            battleStatus->controlScript = script;
+            battleStatus->controlScriptID = script->id;
+            script->owner1.enemyID = 0x100;
+            state->unk_18.x = partner->currentPos.x;
+            state->unk_18.y = 0.0f;
+            state->unk_18.z = partner->currentPos.z;
+            gBattleState2 = 3;
+            break;
+        case 3:
+            if (does_script_exist(battleStatus->controlScriptID) == FALSE) {
```
Comment (nanaian) [inline src/16F740.c:611]:
```suggestion
            if (!does_script_exist(battleStatus->controlScriptID)) {
```

File: src/16F740.c [hunk 11]
```diff
@@ -282,7 +558,132 @@ ApiStatus DisablePartnerBlur(Evt* script, s32 isInitialCall) {
...
+                partner->currentPos.x = player->currentPos.x;
+                partner->currentPos.y = player->currentPos.y + 25.0f;
+                partner->currentPos.z = player->currentPos.z;
+                gBattleState2 = 4;
+            }
+            break;
+        case 4:
+            partner = battleStatus->partnerActor;
+            if (partner != NULL) {
+                if (does_script_exist(partner->takeTurnID) == FALSE) {
```
Comment (nanaian) [inline src/16F740.c:631]:
```suggestion
                if (!does_script_exist(partner->takeTurnID)) {
```

Comment (nanaian) [reply src/16F740.c:631]: etc

File: src/16F740.c [hunk 12]
```diff
@@ -297,7 +698,255 @@ INCLUDE_ASM(s32, "16F740", btl_state_update_end_player_turn);
...
+                        enemyActor->status = 0;
+                    }
+                }
+                func_8024E40C(2);
+                gBattleState2 = 2;
+            }
+            break;
+
+        case 2:
+            for (i = 0; i < 24; i++) {
```
Comment (nanaian) [inline src/16F740.c:746]: 24 should probably be ARRAY_COUNT(battleStatus->enemyActors)

**Review by nanaian — DISMISSED** [review]

File: src/16F740.c [hunk 13]
```diff
@@ -116,15 +120,15 @@ void btl_state_update_switch_to_player(void) {
     s32 i;
 
     if (gBattleState2 == 0) {
-        gBattleStatus.flags1 &= ~0x80000;
+        gBattleStatus.flags1 &= ~BS_FLAGS1_80000;
         reset_actor_turn_info();
-        gBattleStatus.selectedMoveID = 0;
+        gBattleStatus.selectedMoveID = MOVE_NOTHING;
```
Comment (ethteck) [inline src/16F740.c:125]: I think this should be MOVE_NONE (thoughts @nanaian?) - I realize you didn't add it in this PR though

File: src/16F740.c [hunk 14]
```diff
@@ -297,7 +698,256 @@ INCLUDE_ASM(s32, "16F740", btl_state_update_end_player_turn);
...
+                    enemyActor = battleStatus->enemyActors[i];
+                    if (enemyActor != NULL) {
+                        enemyActor->status = 0;
+                    }
+                }
+                func_8024E40C(2);
+                gBattleState2 = 2;
+            }
+            break;
+
```
Comment (ethteck) [inline src/16F740.c:744]: No empty line in between cases needed

**Review by ethteck — CHANGES_REQUESTED** [review]
A couple minor things. Also it'd be nice if we had an enum for gBattleState2... but we can handle that later. Also great job with these long functions!

**Comment by ThreeMustyFears** [comment]
For the enum would BattleStates2 work for a name, because I could start it and replace the instances in this file

File: src/16F740.c (hunk 13 above)
Comment (nanaian) [reply src/16F740.c:125]: I'm ok with that

**Comment by nanaian** [comment]
> For the enum would BattleStates2 work for a name, because I could start it and replace the instances in this file

Yeah that would be awesome! Names are easy to change

File: include/enums.h [hunk 15]
```diff
@@ -1918,7 +1918,7 @@ enum EffectGfxDataFlags {
 };
 
 enum MoveIDs {
-    MOVE_NOTHING                    = 0x00000000,
+    MOVE_NONE                    = 0x00000000,
```
Comment (nanaian) [inline include/enums.h]: Fix the alignment here pls ❤️

**Review by ethteck — APPROVED** [review]
LGTM! thanks :)

**Pull request merged**